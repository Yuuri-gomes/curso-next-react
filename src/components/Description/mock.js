export const DescriptionPropsMock = {
  description: "dummy description",
};
