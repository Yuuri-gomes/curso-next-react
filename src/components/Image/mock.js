export const ImagePropsMock = {
  cover: "img/img.jpg",
  title: "title 1",
};
