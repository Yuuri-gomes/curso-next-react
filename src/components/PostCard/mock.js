export const PostCardPropsMock = {
  title: "title 1",
  description: "Body 1",
  id: 1,
  cover: "img/img.png",
};
