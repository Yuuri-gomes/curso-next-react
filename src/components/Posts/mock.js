export const PostsPropsMock = {
  posts: [
    {
      id: 1,
      title: "Title 1",
      body: "body 1",
      cover: "img/img.jpg",
    },
    {
      id: 2,
      title: "Title 2",
      body: "body 2",
      cover: "img/img.jpg",
    },
    {
      id: 3,
      title: "Title 3",
      body: "body 3",
      cover: "img/img.jpg",
    },
  ],
};
