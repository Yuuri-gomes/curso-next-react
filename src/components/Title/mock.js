export const TitlePropsMock = {
  title: "Some title",
};
