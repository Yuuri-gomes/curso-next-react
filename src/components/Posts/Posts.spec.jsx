import { render, screen } from "@testing-library/react";
import { PostsComponent } from ".";
import { PostsPropsMock } from "./mock";

describe("<Posts />", () => {
  it("should render posts", () => {
    render(<PostsComponent {...PostsPropsMock} />);

    expect(screen.getAllByRole("heading", { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByRole("heading", { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByText(/body/i)).toHaveLength(3);
  });

  it("should match snapshot", () => {
    const { container } = render(<PostsComponent {...PostsPropsMock} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
