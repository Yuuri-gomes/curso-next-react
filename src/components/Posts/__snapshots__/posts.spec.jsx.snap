// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post-card"
  >
    <img
      alt="Title 1"
      src="img/img.jpg"
    />
    <div
      class="post-content"
    >
      <h1>
        Title 1
         
        1
      </h1>
      <p>
        body 1
      </p>
    </div>
  </div>
  <div
    class="post-card"
  >
    <img
      alt="Title 2"
      src="img/img.jpg"
    />
    <div
      class="post-content"
    >
      <h1>
        Title 2
         
        2
      </h1>
      <p>
        body 2
      </p>
    </div>
  </div>
  <div
    class="post-card"
  >
    <img
      alt="Title 3"
      src="img/img.jpg"
    />
    <div
      class="post-content"
    >
      <h1>
        Title 3
         
        3
      </h1>
      <p>
        body 3
      </p>
    </div>
  </div>
</div>
`;
